keps/sig-cli/3805-ssa-default/README.md

```diff
@@ -0,0 +1,520 @@
+flag is now doing the right thing by server-side applying all their
+resources as expected, causing less confusion and risks of messing up.
+
+#### Story 3
+
+User 3 has never used server-side apply and always used the default
+value of `--server-side`. Applying their existing resources will
+continue to work as expected. New resources, on the other hand, will at
+first apply the same but may eventually show-up with a conflict. They
+may have to update scripts to continue to work with
+`--server-side=false` if they want to, or they can update their
+scripts/workflow to address the conflicts properly.
+
+### Risks and Mitigations
+
+This design has no impact on security, but some consequences on UX, since the
+default UX of `kubectl apply` will change:
+- Currently, it can never fail because of conflict, since they are
+  overridden by default. The new default for Server-Side Apply is to
+  fail on conflicts, unless the `--force-conflict` flag is used. While
+  people can re-run the command with the flag, this might impact CI/CD
+  scripts that use `kubectl apply` directly, since they may not have a
+  break-glass way to address that.
+- CSA injects a last-applied-annotation into the objects that it
+  applies, but these don’t make sense in the context of SSA. An API or
+  tool that would use this annotation to detect which resources have
+  been applied would fail to find it for server-side applied objects.
+  Note that doing that is heavily frowned upon.
+- CSA users expect all excluded fields to be removed from the applied
+  object on the server. SSA has more complicated semantics for removing
+  fields (e.g. if another user manages fields)
+
+<!-- Missing mitigation here -->
+
+## Design Details
+
+As mentioned in the proposal, the new value of `auto` will detect if a
+resource has been client-side applied before and will continue to
+client-side apply these resources, while all other resources will be
+server-side applied. This means that new resources will be server-side
+applied by default.
+
+The `last-applied` annotation is used to detect previous client-side
+applied object, hence it is not inserted for server-side objects.
+
+The interaction with other `kubectl apply` flags can be described as
+follows:
+- `--force-conflicts` only applies to existing resources that are
+  server-side applied
+- `--fieldmanager` applies to all resources
+- The `--prune` family of flags, I have no idea
+- `--overwrite` only applies to client-side applied resources
+
+Some other commands might be impacted. `kubectl create` (and family)
+notably have a `--save-config` flag that create the last-applied
+annotation. While I don't know how many people actually use the flag,
+the idea of saving this as a config is confusing, since people don't
+actually have the file and so the situation doesn't really fit well the
+`apply` workflow. We suggest adding a warning when this flag is used, as
+well as updating its documentation to suggest not using it.
```

Comment:
:100: - I'm even inclined to deprecated the flag when doing so, to make it more explicit.